App/features/cafes/sagas.ts

```diff
@@ -24,22 +48,99 @@ function* registerCafe(
 function* deregisterCafe(
   action: ActionType<typeof actions.deregisterCafe.request>
 ) {
-  const { id, success } = action.payload
+  const { peerId, success } = action.payload
   try {
-    yield call(Textile.cafes.deregister, id)
-    yield put(actions.deregisterCafe.success(id))
-    yield put(refreshCafeSessionsRequest())
+    yield call(Textile.cafes.deregister, peerId)
+    yield put(actions.deregisterCafe.success(peerId))
+    yield put(actions.getCafeSessions.request())
     if (success) {
       yield call(success)
     }
   } catch (error) {
-    yield put(actions.deregisterCafe.failure({ id, error }))
+    yield put(actions.deregisterCafe.failure({ peerId, error }))
+  }
+}
+
+function* getCafeSessions() {
+  while (true) {
+    try {
+      yield take(getType(actions.getCafeSessions.request))
+      const list: ICafeSessionList = yield call(Textile.cafes.sessions)
+      yield put(actions.getCafeSessions.success({ sessions: list.items }))
+    } catch (error) {
+      yield call(logNewEvent, 'getCafeSessions', error.message, true)
+      yield put(actions.getCafeSessions.failure({ error }))
+    }
+  }
+}
+
+function* refreshCafeSession(
+  action: ActionType<typeof actions.refreshCafeSession.request>
+) {
+  const { peerId } = action.payload
+  try {
+    const session = yield call(Textile.cafes.refreshSession, peerId)
+    if (session) {
+      yield put(actions.refreshCafeSession.success({ session }))
+    } else {
+      throw new Error('session not found')
+    }
+  } catch (error) {
+    const message =
+      (error.message as string) || (error as string) || 'unknown error'
+    if (message === 'unauthorized') {
+      try {
+        let token: string | undefined
+        // try to get the cafe token from static cafes bundled with the app
+        const cafe = cafesMap[peerId]
+        if (cafe) {
+          token = cafe.token
+        } else {
+          // if not, try to get the token from persisted data about registered cafes
+          const cafe: Cafe = yield select((state: RootState) =>
+            makeCafeForPeerId(peerId)(state.cafes)
+          )
+          token = cafe.token
```

Comment:
Or get the token from the persisted data about registered cafes. The bundled data takes priority in case we ship updated bundled data at a later time than the cafe was registered.